hive-metastore/src/main/java/org/apache/iceberg/hive/HiveMetadataPreservingTableOperations.java

```diff
@@ -180,26 +183,27 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
             baseMetadataLocation, metadataLocation, database, tableName);
       }
 
-      setParameters(newMetadataLocation, tbl, false);
-
-      if (tableExists) {
-        metaClients.run(client -> {
-          EnvironmentContext envContext = new EnvironmentContext(
-              ImmutableMap.of(StatsSetupConst.DO_NOT_UPDATE_STATS, StatsSetupConst.TRUE)
-          );
-          LOG.debug("Updating the metadata location of the following table:");
-          logTable(tbl);
-          LOG.debug("Metadata Location: {}", tbl.getParameters().get(METADATA_LOCATION_PROP));
-          ALTER_TABLE.invoke(client, database, tableName, tbl, envContext);
-          return null;
-        });
-      } else {
-        metaClients.run(client -> {
-          client.createTable(tbl);
-          return null;
-        });
+      // [LINKEDIN] comply to the new signature of setting Hive table's properties by
+      // setting newly added parameters as empty container.
+      setHmsTableParameters(newMetadataLocation, tbl, ImmutableMap.of(),
+          ImmutableSet.of(), false, ImmutableMap.of());
+
+      try {
+        persistTableVerbal(tbl, tableExists);
+        commitStatus = CommitStatus.SUCCESS;
+      } catch (Throwable persistFailure) {
+        LOG.error("Cannot tell if commit to {}.{} succeeded, attempting to reconnect and check.",
+            database, tableName, persistFailure);
+        commitStatus = checkCommitStatus(newMetadataLocation, metadata);
+        switch (commitStatus) {
+          case SUCCESS:
+            break;
+          case FAILURE:
+            throw persistFailure;
+          case UNKNOWN:
+            throw new CommitStateUnknownException(persistFailure);
```

Comment:
does it make sense to log here too, given the log at line 195? 
If you agree, I think that would apply to `hiveTableOperations.java` too